pkg/reconciler/kubernetes/tektondashboard/tektondashboard.go

```diff
@@ -56,81 +67,238 @@ var _ tektondashboardreconciler.Finalizer = (*Reconciler)(nil)
+		}); err != nil {
+		logger.Error("Failed to delete installer set created by TektonDashboard", err)
+		return err
 	}
 
 	if err := r.extension.Finalize(ctx, original); err != nil {
 		logger.Error("Failed to finalize platform resources", err)
 	}
-	logger.Info("Deleting cluster-scoped resources")
-	manifest, err := r.installed(ctx, original)
-	if err != nil {
-		logger.Error("Unable to fetch installed manifest; no cluster-scoped resources will be finalized", err)
-		return nil
-	}
-	if err := common.Uninstall(ctx, manifest, nil); err != nil {
-		logger.Error("Failed to finalize platform resources", err)
-	}
 	return nil
 }
 
 // ReconcileKind compares the actual state with the desired, and attempts to
 // converge the two.
-func (r *Reconciler) ReconcileKind(ctx context.Context, tt *v1alpha1.TektonDashboard) pkgreconciler.Event {
+func (r *Reconciler) ReconcileKind(ctx context.Context, td *v1alpha1.TektonDashboard) pkgreconciler.Event {
 	logger := logging.FromContext(ctx)
-	tt.Status.InitializeConditions()
-	tt.Status.ObservedGeneration = tt.Generation
+	td.Status.InitializeConditions()
+	td.Status.ObservedGeneration = td.Generation
+
+	logger.Infow("Reconciling TektonDashboards", "status", td.Status)
 
-	logger.Infow("Reconciling TektonDashboards", "status", tt.Status)
+	r.releaseVersion = common.TargetVersion(td)
 
-	if tt.GetName() != watchedResourceName {
+	if td.GetName() != watchedResourceName {
 		msg := fmt.Sprintf("Resource ignored, Expected Name: %s, Got Name: %s",
 			watchedResourceName,
-			tt.GetName(),
+			td.GetName(),
 		)
 		logger.Error(msg)
-		tt.GetStatus().MarkInstallFailed(msg)
+		td.GetStatus().MarkInstallFailed(msg)
 		return nil
 	}
 
 	// find the valid tekton-pipeline installation
 	if _, err := common.PipelineReady(r.pipelineInformer); err != nil {
 		if err.Error() == common.PipelineNotReady {
-			tt.Status.MarkDependencyInstalling("tekton-pipelines is still installing")
+			td.Status.MarkDependencyInstalling("tekton-pipelines is still installing")
 			// wait for pipeline status to change
 			return fmt.Errorf(common.PipelineNotReady)
 		}
 		// (tektonpipeline.opeator.tekton.dev instance not available yet)
-		tt.Status.MarkDependencyMissing("tekton-pipelines does not exist")
+		td.Status.MarkDependencyMissing("tekton-pipelines does not exist")
 		return err
 	}
-	tt.Status.MarkDependenciesInstalled()
+	td.Status.MarkDependenciesInstalled()
 
-	if err := r.extension.PreReconcile(ctx, tt); err != nil {
+	if err := r.extension.PreReconcile(ctx, td); err != nil {
+		td.Status.MarkPreReconcilerFailed(fmt.Sprintf("PreReconciliation failed: %s", err.Error()))
 		return err
 	}
-	stages := common.Stages{
-		common.AppendTarget,
-		r.transform,
-		common.Install,
-		common.CheckDeployments,
-	}
+
+	// Mark PreReconcile Complete
+	td.Status.MarkPreReconcilerComplete()
+
 	manifest := r.manifest.Append()
-	return stages.Execute(ctx, &manifest, tt)
+
+	if err := common.AppendTarget(ctx, &manifest, td); err != nil {
```

Comment:
is it possible to do this in controller,go?

Reply:
we can read both the release yaml in 2 separate fields and use that

Reply:
Could you please check once now ?